src/core/filesystem.cc

```diff
@@ -1033,6 +1004,70 @@ ASFileSystem::WriteTextFile(
   }
   return Status::Success;
 }
+
+
+Status 
+CreateASFileSystem(const std::string& path, FileSystem** file_system){
+  re2::RE2 regex(AS_URL_PATTERN);
```

Comment:
is it a convention in this repo to parse urls with regex? do we have a utility like gurl?